app/controllers/support/providers/copy_courses_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Support
+  module Providers
+    class CopyCoursesController < SupportController
+      before_action :recruitment_cycle
+
+      def new
+        @target_provider = Provider.find(params[:provider_id])
+        @copy_courses_form = CopyCoursesForm.new(@target_provider)
+      end
+
+      def create
+        @target_provider = Provider.find(params[:provider_id])
+        @copy_courses_form = CopyCoursesForm.new(@target_provider, recruitment_cycle.providers.find_by(provider_code: params[:course][:autocompleted_provider_code]))
+
+        if @copy_courses_form.valid?
+          sites_copy_to_course = params[:schools] ? Sites::CopyToCourseService : ->(*) {}
+
+          copier = ::Courses::CopyToProviderService.new(sites_copy_to_course:, enrichments_copy_to_course: Enrichments::CopyToCourseService.new, force: true)
+
+          Provider.transaction do
+            @copy_courses_form.provider.courses.map do |course|
```

Comment:
On the PR description it mentions "Copy all courses or some courses" but here is all courses. Can we remove the "some courses" from description just to avoid confusion?

I assume the copying specific courses would be done on a later interaction when we will have a support ticket?

Reply:
Yeah I've not implemented "some courses" because unwanted courses can be deleted